### PR TITLE
Add PortraitDepth API and infrastructure

### PR DESCRIPTION
**Pull request by ahmedsabie**
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/1005)

### CODE REVIEW

File: depth-estimation/karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+const karmaTypescriptConfig = {
+  tsconfig: 'tsconfig.test.json',
+  // Disable coverage reports and instrumentation by default for tests
```
Comment (ruofeidu) [inline depth-estimation/karma.conf.js:20]: nit: for tests.

File: depth-estimation/karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  coverageOptions: {instrumentation: false},
+  reports: {},
+  bundlerOptions: {
+    sourceMap: true,
+    // Process any non es5 code through karma-typescript-es6-transform (babel)
```
Comment (ruofeidu) [inline depth-estimation/karma.conf.js:25]: nit: Process any non ES5 code through karma-typescript-es6-transform (babel).

File: depth-estimation/karma.conf.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    acornOptions: {ecmaVersion: 8},
+    transforms: [
+      require('karma-typescript-es6-transform')({
+        presets: [
+          // ensure we get es5 by adding IE 11 as a target
```
Comment (ruofeidu) [inline depth-estimation/karma.conf.js:30]: nit: Ensure we get ES5 by adding IE 11 as a target.

**Review by lina128 — APPROVED** [review]
Thank you Ahmed!

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-:-N0qXKYQ6u7lDlUExL7n:bfpkus8)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @ahmedsabie and @ruofeidu)

___
*[`depth-estimation/README.md` line 3 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qVTGf1q9Lq_C3xijy:-N0qVTGf1q9Lq_C3xijz:boucjj1) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/README.md#L3)):*
> ```Markdown
> # Depth Estimation
> 
> This package provides models for running depth estimation in TensorFlow.js.
> ```

Add one sentence here:

Currently, we provide 1 model option:



_Code quote:_
```Markdown
This package provides models for running depth estimation in TensorFlow.js.
```
___
*[`depth-estimation/README.md` line 5 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qVvNe5C2vFuiKbQ1Q:-N0qVvNe5C2vFuiKbQ1R:b-tpujl0) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/README.md#L5)):*
> ```Markdown
> This package provides models for running depth estimation in TensorFlow.js.
> 
> ## Portrait Depth API
> ```

#### ARPortraitDepth



_Code quote:_
```Markdown
## Portrait Depth API
```
___
*[`depth-estimation/README.md` line 18 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qWIu3BRHh0NSvo3Z0:-N0qWIu3BRHh0NSvo3Z1:b-3fnx69) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/README.md#L18)):*
> ```Markdown
> ## Table of Contents
> 
> - [Depth Estimation](#depth-estimation)
> ```

Remove this line.



_Code quote:_
```Markdown
- [Depth Estimation](#depth-estimation)
```
___
*[`depth-estimation/README.md` line 19 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qWKQ734opBKb9MY5J:-N0qWKQ734opBKb9MY5K:b-3fnx69) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/README.md#L19)):*
> ```Markdown
> 
> - [Depth Estimation](#depth-estimation)
>   - [Portrait Depth API](#portrait-depth-api)
> ```

Remove this line.



_Code quote:_
```Markdown
  - [Portrait Depth API](#portrait-depth-api)
```
___
*[`depth-estimation/README.md` line 20 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qWMCm1xXLtnqJPu2g:-N0qWMCm1xXLtnqJPu2h:b-3fnx69) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/README.md#L20)):*
> ```Markdown
> - [Depth Estimation](#depth-estimation)
>   - [Portrait Depth API](#portrait-depth-api)
>   - [Table of Contents](#table-of-contents)
> ```

Remove this line.



_Code quote:_
```Markdown
  - [Table of Contents](#table-of-contents)
```
___
*[`depth-estimation/README.md` line 31 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qWTcm5RUaI_qpK-LP:-N0qWTcm5RUaI_qpK-LQ:byp18fd) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/README.md#L31)):*
> ```Markdown
> 
> First, you create an estimator by choosing one of the models from
> `SupportedModels`, including `PortraitDepth`.
> ```

Remove



_Code quote:_
```Markdown
, including `PortraitDepth`
```
___
*[`depth-estimation/README.md` line 36 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qWVJr4K_ZF4HoLzzY:-N0qWVJr4K_ZF4HoLzzZ:bjf5ifm) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/README.md#L36)):*
> ```Markdown
> 
> ```javascript
> const model = depthEstimation.SupportedModels.PortraitDepth;
> ```

ARPortraitDepth



_Code quote:_
```Markdown
PortraitDepth
```
___
*[`depth-estimation/README.md` line 71 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qWox_AOT012dTii-N:-N0qWox_AOT012dTii-O:bjf5ifm) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/README.md#L71)):*
> ```Markdown
> and their performance.
> 
> [PortraitDepth Documentation](TODO: add link)
> ```

ARPortraitDepth



_Code quote:_
```Markdown
PortraitDepth
```
___
*[`depth-estimation/src/portrait_depth/README.md` line 1 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0p4KfpGD4bsCIhxxx0-r1-1:-N0qWyYX1jRn74evNrWM:bjf5ifm) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/src/portrait_depth/README.md#L1)):*
<details><summary><i>Previously, ruofeidu wrote…</i></summary><blockquote>

nit: Portrait Depth
</blockquote></details>

ARPortraitDepth



_Code quote:_
```Markdown
PortraitDepth
```
___
*[`depth-estimation/src/portrait_depth/README.md` line 74 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qXF5w15tZSu_0HNn_:-N0qXF5w15tZSu_0HNna:bjf5ifm) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/src/portrait_depth/README.md#L74)):*
> ```Markdown
> 
> ```javascript
> const model = depthEstimation.SupportedModels.PortraitDepth;
> ```

ARPortraitDepth



_Code quote:_
```Markdown
PortraitDepth
```

**Review by ruofeidu — APPROVED** [review]

**Review by ahmedsabie — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/1005)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @lina128)

___
*[`depth-estimation/README.md` line 3 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qVTGf1q9Lq_C3xijy:-N0rXxxBB8Mw3sksm0Hk:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/README.md#L3)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

Add one sentence here:

Currently, we provide 1 model option:
</blockquote></details>

Done.
___
*[`depth-estimation/README.md` line 5 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qVvNe5C2vFuiKbQ1Q:-N0rXydECTTsmUovAcXN:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/README.md#L5)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

#### ARPortraitDepth
</blockquote></details>

Done.
___
*[`depth-estimation/README.md` line 18 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qWIu3BRHh0NSvo3Z0:-N0rXzbB1hPS5AFXPgB3:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/README.md#L18)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

Remove this line.
</blockquote></details>

Done.
___
*[`depth-estimation/README.md` line 19 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qWKQ734opBKb9MY5J:-N0rY-3A2Mw3eyQr_PgL:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/README.md#L19)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

Remove this line.
</blockquote></details>

Done.
___
*[`depth-estimation/README.md` line 20 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qWMCm1xXLtnqJPu2g:-N0rY-dY37YIiR8-Melk:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/README.md#L20)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

Remove this line.
</blockquote></details>

Done.
___
*[`depth-estimation/README.md` line 31 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qWTcm5RUaI_qpK-LP:-N0rY02pDh1vndJi2CEG:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/README.md#L31)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

Remove
</blockquote></details>

Done.
___
*[`depth-estimation/README.md` line 36 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qWVJr4K_ZF4HoLzzY:-N0rY0ul1wiP3GeeXgI7:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/README.md#L36)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

ARPortraitDepth
</blockquote></details>

Done.
___
*[`depth-estimation/README.md` line 71 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qWox_AOT012dTii-N:-N0rY1Rt2FCjYyAbFifC:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/README.md#L71)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

ARPortraitDepth
</blockquote></details>

Done.
___
*[`depth-estimation/src/portrait_depth/README.md` line 74 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/1005#-N0qXF5w15tZSu_0HNn_:-N0rY3YpCcXJkWsT8ROp:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/6fbbe4ddf8fdd34af26414fdce813069043b6280/depth-estimation/src/portrait_depth/README.md#L74)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

ARPortraitDepth
</blockquote></details>

Done.

**Pull request merged**